src/Interval.h

```diff
@@ -0,0 +1,301 @@
+/*
+ * LSST Data Management System
+ * Copyright 2014-2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_SG_INTERVAL_H_
+#define LSST_SG_INTERVAL_H_
+
+/// \file
+/// \author Serge Monkewitz
+/// \brief This file defines a template representing closed real intervals.
+
+#include <algorithm>
+#include <ostream>
+
+#include "SpatialRelation.h"
+
+
+namespace lsst {
+namespace sg {
+
+/// `Interval` represents a closed interval of the real numbers
+/// by its upper and lower bounds. It is parameterized by the Scalar
+/// type, which must be constructible from a double.
+///
+/// An interval with identical upper and lower bounds contains
+/// a single point, and is equal to that scalar bound.
+///
+/// An interval with an upper bound less than its lower bound is
+/// empty (contains no points), as are intervals with NaN as
+/// either bound.
```

Comment:
Does Interval behave appropriately when either or both bounds are infinite?  If not, it might be worth calling that out explicitly in the documentation.